lib/generateNormals.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+var readAccessor = require('./readAccessor');
+var gltfPrimitiveToCesiumGeometry = require('./gltfPrimitiveToCesiumGeometry');
+var cesiumGeometryToGltfPrimitive = require('./cesiumGeometryToGltfPrimitive');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var WebGLConstants = Cesium.WebGLConstants;
+var GeometryPipeline = Cesium.GeometryPipeline;
+
+module.exports = generateNormals;
+
+
+/**
+ * Generates normals for primitives if they do not exist.
+ *
+ * @param gltf
+ * @param {boolean} isHard
+ */
+function generateNormals(gltf) {
+    var generatedNormals = false;
+    var primitives = getPrimitives(gltf);
+    var primitivesLength = primitives.length;
+    for (var i = 0; i < primitivesLength; i++) {
+        var primitive = primitives[i];
+        var positionAccessorId = primitive.attributes.POSITION;
+        var normalAccessorId = primitive.attributes.NORMAL;
+        var indicesAccessorId = primitive.indices;
+        var positionAccessor = gltf.accessors[positionAccessorId];
+        var position = [];
+        readAccessor(gltf, positionAccessor, position);
+
+        var options = {
+            attributeLength : position.length
+        };
```

Comment:
This options thing is somewhat confusing, and it would be better is `cesiumGeometryToGltfPrimitive` didn't need to use it at all. Can't it get the attribute length from the Cesium geometry?